lib/index.js

```diff
@@ -370,14 +426,28 @@ exports.getStorageClient = (ctx) => {
     const url = `${ctx.configuration.fusebit_storage_audience}/v1/account/${ctx.configuration.fusebit_storage_account_id}/subscription/${ctx.configuration.fusebit_storage_subscription_id}/storage/${ctx.configuration.fusebit_storage_id}`;
 
     return {
+        etag: null,
         get: async () => {
             const response = await Superagent.get(url)
                 .set('Authorization', `Bearer ${accessToken}`)
                 .ok((res) => res.status < 300 || res.status === 404);
+            this.etag = response.headers.etag;
             return response.status === 404 ? undefined : response.body.data;
         },
-        put: async (data) => {
-            await Superagent.put(url).set('Authorization', `Bearer ${accessToken}`).send({ data });
+        put: async (data, oldEtag) => {
```

Comment:
Isn't the idea of the code above to store the etag on the object returned from this function? Why does it need to be passed in?

Reply:
It was between passing in the etag or some kind of "force" boolean flag to tell it to overwrite even if the stored internal etag is wrong. Now that you raised it, the latter does seem like a nicer OM, I've switched to that.